unetsocket/c/Makefile

```diff
@@ -38,28 +45,89 @@ rxdata.o: samples/rxdata.c unet.h fjage.h
+
+txsignal.o: samples/txsignal.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/txsignal.c -o samples/txsignal.o
+
+txsignal: txsignal.o
+	$(CC) -o samples/txsignal unet_ext.o samples/txsignal.o libunet.a libfjage.a -lpthread -lm
+
+pbrecord.o: samples/pbrecord.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/pbrecord.c -o samples/pbrecord.o
+
+pbrecord: pbrecord.o unet_ext.o
+	$(CC) -o samples/pbrecord unet_ext.o samples/pbrecord.o libunet.a libfjage.a -lpthread -lm
+
+bbrecord.o: samples/bbrecord.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/bbrecord.c -o samples/bbrecord.o
+
+bbrecord: bbrecord.o
+	$(CC) -o samples/bbrecord unet_ext.o samples/bbrecord.o libunet.a libfjage.a -lpthread -lm
 
-test: test_unet
+npulses.o: samples/npulses.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/npulses.c -o samples/npulses.o
+
+npulses: npulses.o unet_ext.o
+	$(CC) -o samples/npulses unet_ext.o samples/npulses.o libunet.a libfjage.a -lpthread -lm
+
+wakeup.o: samples/wakeup.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/wakeup.c -o samples/wakeup.o
+
+wakeup: wakeup.o unet_ext.o
+	$(CC) -o samples/wakeup unet_ext.o samples/wakeup.o libunet.a libfjage.a -lpthread -lm
+
+rs232_wakeup.o: samples/rs232_wakeup.c unet.h fjage.h unet_ext.o
+	$(CC) $(CFLAGS) -c samples/rs232_wakeup.c -o samples/rs232_wakeup.o
+
+rs232_wakeup: rs232_wakeup.o unet_ext.o
+	$(CC) -o samples/rs232_wakeup unet_ext.o samples/rs232_wakeup.o libunet.a libfjage.a -lpthread -lm
+
+test: test_unet unet_ext.o
 	rm -rf *.zip
 	rm -rf fjage-*
 
-test_unet.o: test/test_unet.c unet.h fjage.h
+test_unet.o: test/test_unet.c unet.h fjage.h unet_ext.o
 	$(CC) $(CFLAGS) -c test/test_unet.c -o test/test_unet.o
 
 test_unet: test_unet.o
-	$(CC) -o test/test_unet test/test_unet.o libunet.a libfjage.a -lpthread -lm
+	$(CC) -o test/test_unet unet_ext.o test/test_unet.o libunet.a libfjage.a -lpthread -lm
 
 clean:
-	rm -rf c-api libunet.a *.o samples/*.o test/*.o test/test_unet samples/wakeup samples/rs232_wakeup samples/txdata samples/rxdata samples/range samples/setpowerlevel
+	rm -rf c-api libunet.a *.o samples/*.o test/*.o test/test_unet samples/wakeup samples/rs232_wakeup samples/txdata samples/rxdata samples/range samples/setpowerlevel samples/txsignal samples/pbrecord samples/recordedsignal.txt samples/bbrecord samples/npulses samples/gpio
 	rm -rf "$(FJAGE_DIR)" "$(FJAGE_VER).zip" fjage.h libfjage.a kissfft kiss_fft.h fjage
+	rm -rf $(BUILD)
+
+$(CONTRIB_DIR): $(BUILD_API)
+	curl -LO https://github.com/org-arl/fjage/archive/$(FJAGE_VER).zip
+	mv $(FJAGE_VER).zip $(BUILD)/
+	cd $(BUILD) && unzip -a $(FJAGE_VER).zip
+	mkdir -p $(BUILD_API)/libs/fjage/gateways
+	cp -r $(BUILD)/$(FJAGE_DIR)/gateways/c $(BUILD_API)/libs/fjage/gateways/c
+	curl -s -L https://github.com/org-arl/unet-contrib/archive/$(CONTRIB_VER).zip --output $(BUILD)/unet-contrib-$(CONTRIB_VER).zip
```

Comment:
Why do we need to download the for the source for `unet-contrib` inside the directory with the source code for `unet-contrib` ?

Reply:
This is required when making a distribution package. This downloads`unet-contrib` at a specific commit hash (given by `CONTRIB_VER` in Makefile). 
With this even if I am in some different branch say `dev`, I should be able to say `make package` and it will download the correct source from `unet-contrib` and build the package.

Reply:
Hmm? Why if I'm on a specific branch then it means I want to build from that branch. If I want a specific commit, I will git checkout <commit> and do a `make`.

Reply:
Sure. Makes sense. I have updated the Makefile to fix this now in bd95ed77ef7c90edbba04dd67f8fb235262e08c8